File: Backend-main/src/contact/contact.controller.ts
import { Body, Controller, Post } from '@nestjs/common';
import { ContactFormRequest } from './interface/contact.interface';
import { ContactService } from './contact.service';
import { PublicWithOptionalSession } from 'src/guard/optional-session.guard';
import { Session } from 'supertokens-nestjs';
import { SessionContainer } from 'supertokens-node/recipe/session';

@Controller('contact')
export class ContactController {
  constructor(private readonly contactService: ContactService) {}

  @Post()
  @PublicWithOptionalSession()
  async createContactForm(
    @Body() request: ContactFormRequest,
    @Session() session?: SessionContainer,
  ): Promise<void> {
    const userId = session?.getUserId();
    await this.contactService.createContactFormEntry(request, userId);
  }
}

File: spring-backend/src/main/java/com/simmortal/contact/ContactController.java
package com.simmortal.contact;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("contact")
public class ContactController {
  private final ContactService contactService;

  public ContactController(ContactService contactService) {
    this.contactService = contactService;
  }

  @PostMapping
  public void createContactForm(
      @RequestBody ContactFormRequest request,
      @RequestHeader(value = "X-User-Id", required = false) String userId
  ) {
    contactService.createContactFormEntry(request, userId);
  }
}
